tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -1294,21 +1313,35 @@ class RunWskRestCmd() extends FlatSpec with RunWskCmd with Matchers with ScalaFu
 
   def convertMapIntoKeyValue(params: Map[String, JsValue],
                              feed: Option[String] = None,
-                             web: Option[String] = None): JsArray = {
+                             web: Option[String] = None,
+                             oldParams: Vector[JsObject] = Vector[JsObject]()): JsArray = {
     var paramsList = Vector[JsObject]()
     params foreach { case (key, value) => paramsList :+= JsObject("key" -> key.toJson, "value" -> value.toJson) }
     paramsList = feed map { f =>
       paramsList :+ JsObject("key" -> "feed".toJson, "value" -> f.toJson)
     } getOrElse paramsList
+
+    if (paramsList.size == 0) {
+      paramsList = oldParams
+    }
```

Comment:
Please remove mutability here. You don't need it, your code is even almost there.

```scala
val newParams = 
  params
    .map { case (key, value) => JsObject("key" -> key.toJson, "value" -> value) } 
    ++ feed.map(f => JsObject("key" -> "feed".toJson, "value" -> f.toJson)

val paramsList = if(newParams.nonEmpty) newParams else oldParams
```

That removes any possibility of breaking the code accidentally by trying to change those variables again.

Reply:
The paramsList is not Vector[JsObject](),  which is unable to convert to JsArray.